Traceback (most recent call last):
  File "/home/runner/.local/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/runner/.local/lib/python3.10/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/runner/.local/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/home/runner/.local/lib/python3.10/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/home/runner/.local/lib/python3.10/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/runner/.local/lib/python3.10/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
%load_ext fortranmagic
------------------

----- stderr -----
/home/runner/.local/lib/python3.10/site-packages/fortranmagic.py:195: UserWarning: This is now an optional IPython functionality, using fortranmagic_cache requires you to install the `pickleshare` library.
  cdir = self.shell.db['fortranmagic_cache']
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mTypeError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[8], line 1[0m
[0;32m----> 1[0m [43mget_ipython[49m[43m([49m[43m)[49m[38;5;241;43m.[39;49m[43mrun_line_magic[49m[43m([49m[38;5;124;43m'[39;49m[38;5;124;43mload_ext[39;49m[38;5;124;43m'[39;49m[43m,[49m[43m [49m[38;5;124;43m'[39;49m[38;5;124;43mfortranmagic[39;49m[38;5;124;43m'[39;49m[43m)[49m

File [0;32m~/.local/lib/python3.10/site-packages/IPython/core/interactiveshell.py:2480[0m, in [0;36mInteractiveShell.run_line_magic[0;34m(self, magic_name, line, _stack_depth)[0m
[1;32m   2478[0m     kwargs[[38;5;124m'[39m[38;5;124mlocal_ns[39m[38;5;124m'[39m] [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mget_local_scope(stack_depth)
[1;32m   2479[0m [38;5;28;01mwith[39;00m [38;5;28mself[39m[38;5;241m.[39mbuiltin_trap:
[0;32m-> 2480[0m     result [38;5;241m=[39m [43mfn[49m[43m([49m[38;5;241;43m*[39;49m[43margs[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m
[1;32m   2482[0m [38;5;66;03m# The code below prevents the output from being displayed[39;00m
[1;32m   2483[0m [38;5;66;03m# when using magics with decorator @output_can_be_silenced[39;00m
[1;32m   2484[0m [38;5;66;03m# when the last Python token in the expression is a ';'.[39;00m
[1;32m   2485[0m [38;5;28;01mif[39;00m [38;5;28mgetattr[39m(fn, magic[38;5;241m.[39mMAGIC_OUTPUT_CAN_BE_SILENCED, [38;5;28;01mFalse[39;00m):

File [0;32m~/.local/lib/python3.10/site-packages/IPython/core/magics/extension.py:33[0m, in [0;36mExtensionMagics.load_ext[0;34m(self, module_str)[0m
[1;32m     31[0m [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m module_str:
[1;32m     32[0m     [38;5;28;01mraise[39;00m UsageError([38;5;124m'[39m[38;5;124mMissing module name.[39m[38;5;124m'[39m)
[0;32m---> 33[0m res [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mshell[49m[38;5;241;43m.[39;49m[43mextension_manager[49m[38;5;241;43m.[39;49m[43mload_extension[49m[43m([49m[43mmodule_str[49m[43m)[49m
[1;32m     35[0m [38;5;28;01mif[39;00m res [38;5;241m==[39m [38;5;124m'[39m[38;5;124malready loaded[39m[38;5;124m'[39m:
[1;32m     36[0m     [38;5;28mprint[39m([38;5;124m"[39m[38;5;124mThe [39m[38;5;132;01m%s[39;00m[38;5;124m extension is already loaded. To reload it, use:[39m[38;5;124m"[39m [38;5;241m%[39m module_str)

File [0;32m~/.local/lib/python3.10/site-packages/IPython/core/extensions.py:62[0m, in [0;36mExtensionManager.load_extension[0;34m(self, module_str)[0m
[1;32m     55[0m [38;5;250m[39m[38;5;124;03m"""Load an IPython extension by its module name.[39;00m
[1;32m     56[0m 
[1;32m     57[0m [38;5;124;03mReturns the string "already loaded" if the extension is already loaded,[39;00m
[1;32m     58[0m [38;5;124;03m"no load function" if the module doesn't have a load_ipython_extension[39;00m
[1;32m     59[0m [38;5;124;03mfunction, or None if it succeeded.[39;00m
[1;32m     60[0m [38;5;124;03m"""[39;00m
[1;32m     61[0m [38;5;28;01mtry[39;00m:
[0;32m---> 62[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_load_extension[49m[43m([49m[43mmodule_str[49m[43m)[49m
[1;32m     63[0m [38;5;28;01mexcept[39;00m [38;5;167;01mModuleNotFoundError[39;00m:
[1;32m     64[0m     [38;5;28;01mif[39;00m module_str [38;5;129;01min[39;00m BUILTINS_EXTS:

File [0;32m~/.local/lib/python3.10/site-packages/IPython/core/extensions.py:79[0m, in [0;36mExtensionManager._load_extension[0;34m(self, module_str)[0m
[1;32m     77[0m     mod [38;5;241m=[39m import_module(module_str)
[1;32m     78[0m mod [38;5;241m=[39m sys[38;5;241m.[39mmodules[module_str]
[0;32m---> 79[0m [38;5;28;01mif[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_call_load_ipython_extension[49m[43m([49m[43mmod[49m[43m)[49m:
[1;32m     80[0m     [38;5;28mself[39m[38;5;241m.[39mloaded[38;5;241m.[39madd(module_str)
[1;32m     81[0m [38;5;28;01melse[39;00m:

File [0;32m~/.local/lib/python3.10/site-packages/IPython/core/extensions.py:129[0m, in [0;36mExtensionManager._call_load_ipython_extension[0;34m(self, mod)[0m
[1;32m    127[0m [38;5;28;01mdef[39;00m [38;5;21m_call_load_ipython_extension[39m([38;5;28mself[39m, mod):
[1;32m    128[0m     [38;5;28;01mif[39;00m [38;5;28mhasattr[39m(mod, [38;5;124m'[39m[38;5;124mload_ipython_extension[39m[38;5;124m'[39m):
[0;32m--> 129[0m         [43mmod[49m[38;5;241;43m.[39;49m[43mload_ipython_extension[49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mshell[49m[43m)[49m
[1;32m    130[0m         [38;5;28;01mreturn[39;00m [38;5;28;01mTrue[39;00m

File [0;32m~/.local/lib/python3.10/site-packages/fortranmagic.py:547[0m, in [0;36mload_ipython_extension[0;34m(ip)[0m
[1;32m    545[0m [38;5;28;01mdef[39;00m [38;5;21mload_ipython_extension[39m(ip):
[1;32m    546[0m [38;5;250m    [39m[38;5;124;03m"""Load the extension in IPython."""[39;00m
[0;32m--> 547[0m     [43mip[49m[38;5;241;43m.[39;49m[43mregister_magics[49m[43m([49m[43mFortranMagics[49m[43m)[49m
[1;32m    549[0m     [38;5;66;03m# enable fortran highlight[39;00m
[1;32m    550[0m     patch [38;5;241m=[39m [38;5;124m"""[39m
[1;32m    551[0m [38;5;124m        if(typeof IPython === [39m[38;5;124m'[39m[38;5;124mundefined[39m[38;5;124m'[39m[38;5;124m) [39m[38;5;124m{[39m
[1;32m    552[0m [38;5;124m            console.log([39m[38;5;124m'[39m[38;5;124mfortranmagic.py: TDOO: JupyterLab [39m[38;5;124m'[39m[38;5;124m +[39m
[0;32m   (...)[0m
[1;32m    557[0m [38;5;124m        }[39m
[1;32m    558[0m [38;5;124m        [39m[38;5;124m"""[39m

File [0;32m~/.local/lib/python3.10/site-packages/IPython/core/magic.py:456[0m, in [0;36mMagicsManager.register[0;34m(self, *magic_objects)[0m
[1;32m    452[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m([38;5;124m"[39m[38;5;124mClass of magics [39m[38;5;132;01m%r[39;00m[38;5;124m was constructed without [39m[38;5;124m"[39m
[1;32m    453[0m                      [38;5;124m"[39m[38;5;124mthe @register_magics class decorator[39m[38;5;124m"[39m)
[1;32m    454[0m [38;5;28;01mif[39;00m [38;5;28misinstance[39m(m, [38;5;28mtype[39m):
[1;32m    455[0m     [38;5;66;03m# If we're given an uninstantiated class[39;00m
[0;32m--> 456[0m     m [38;5;241m=[39m [43mm[49m[43m([49m[43mshell[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mshell[49m[43m)[49m
[1;32m    458[0m [38;5;66;03m# Now that we have an instance, we can register it and update the[39;00m
[1;32m    459[0m [38;5;66;03m# table of callables[39;00m
[1;32m    460[0m [38;5;28mself[39m[38;5;241m.[39mregistry[m[38;5;241m.[39m[38;5;18m__class__[39m[38;5;241m.[39m[38;5;18m__name__[39m] [38;5;241m=[39m m

File [0;32m~/.local/lib/python3.10/site-packages/fortranmagic.py:227[0m, in [0;36mFortranMagics.__init__[0;34m(self, shell)[0m
[1;32m    225[0m [38;5;28mself[39m[38;5;241m.[39m_reloads [38;5;241m=[39m {}
[1;32m    226[0m [38;5;28mself[39m[38;5;241m.[39m_code_cache [38;5;241m=[39m {}
[0;32m--> 227[0m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_cache_open[49m[43m([49m[43m)[49m

File [0;32m~/.local/lib/python3.10/site-packages/fortranmagic.py:196[0m, in [0;36mFortranMagics._cache_open[0;34m(self)[0m
[1;32m    194[0m [38;5;28;01mtry[39;00m:
[1;32m    195[0m     cdir [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mshell[38;5;241m.[39mdb[[38;5;124m'[39m[38;5;124mfortranmagic_cache[39m[38;5;124m'[39m]
[0;32m--> 196[0m     [38;5;28;01mif[39;00m [43mos[49m[38;5;241;43m.[39;49m[43mpath[49m[38;5;241;43m.[39;49m[43misdir[49m[43m([49m[43mcdir[49m[43m)[49m:
[1;32m    197[0m         [38;5;28mself[39m[38;5;241m.[39m_lib_dir [38;5;241m=[39m cdir
[1;32m    198[0m         [38;5;28;01mreturn[39;00m

File [0;32m/usr/lib/python3.10/genericpath.py:42[0m, in [0;36misdir[0;34m(s)[0m
[1;32m     40[0m [38;5;250m[39m[38;5;124;03m"""Return true if the pathname refers to an existing directory."""[39;00m
[1;32m     41[0m [38;5;28;01mtry[39;00m:
[0;32m---> 42[0m     st [38;5;241m=[39m [43mos[49m[38;5;241;43m.[39;49m[43mstat[49m[43m([49m[43ms[49m[43m)[49m
[1;32m     43[0m [38;5;28;01mexcept[39;00m ([38;5;167;01mOSError[39;00m, [38;5;167;01mValueError[39;00m):
[1;32m     44[0m     [38;5;28;01mreturn[39;00m [38;5;28;01mFalse[39;00m

[0;31mTypeError[0m: stat: path should be string, bytes, os.PathLike or integer, not NoneType

